packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2745,7 +2744,7 @@ class Navigator extends StatefulWidget {
   /// has any [Navigator.observers], they will be notified as well (see
   /// [NavigatorObserver.didRemove]). The removed route is disposed without
   /// being notified. The future that had been returned from pushing that routes
-  /// will not complete.
+  /// will not complete. To complete this future, you must call [Route.didComplete] just after.
```

Comment:
I don't think this is really a good idea to ask people to call this manually. I would rather this is called automatically upon remove, or have some special way to listen for remove.

Reply:
I'm afraid of putting it inside : I didn't know the navigator's internal system well.

Reply:
I think if there is a red flag, the ci will catch it